### PR TITLE
fix: solve the problem of incorrect execution time of cron

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What type of this PR
/kind bug

#### What this PR does / why we need it:
fix: solve the problem of incorrect execution time of cron

#### Which issue(s) this PR fixes:
[【流水线】定时任务执行时间在某一些情况下执行时间不正确](https://erda.cloud/erda/dop/projects/387/issues/all?id=210903&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIl0sImFzc2lnbmVlSURzIjpbIjEwMDEyNjEiXX0%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=506&type=BUG)

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |       解决定时任务执行时间不正确问题       |
| 🇨🇳 中文    |     solve the problem of incorrect execution time of cron         |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkg/cron/cron.go [hunk 1]
```diff
@@ -344,6 +344,7 @@ func (c *Cron) run() {
 			// and stop requests.
 			timer = time.NewTimer(100000 * time.Hour)
 		} else {
+			now = c.now()
```
Comment (sfwn) [inline pkg/cron/cron.go:347]: add sample for this change.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Comment by sfwn** [comment]
Check the version this bug imported.
And fixed to these affected versions.

File: pkg/cron/cron.go (hunk 1 above)
Comment (littlejiancc) [reply pkg/cron/cron.go:351]: done

**Review by sfwn — DISMISSED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**